cmake/FindPostgreSQL.cmake

```diff
@@ -51,7 +51,9 @@
 # License for the specific language governing permissions and limitations under
 # the License.
 
-include(${CMAKE_SOURCE_DIR}/cmake/OpenSSL.cmake)
+get_filename_component(PRESENT_DIR "${CMAKE_CURRENT_LIST_FILE}" DIRECTORY)
```

Comment:
Can this be achieved with [CMAKE_CURRENT_LIST_DIR](https://cmake.org/cmake/help/latest/variable/CMAKE_CURRENT_LIST_DIR.html)

Reply:
Since `FindPostgreSQL.cmake` is called by `CMakeLists.txt` files located in different directories (root, root/extensions/*), `CMAKE_CURRENT_LIST_DIR` will provide the corresponding directory location.  

No matter which CMakeLists.txt file calls `FindPostgreSQL.cmake`, we need to include `OpenSSL.cmake` and we know where it is located. So, I am not sure why we would use `CMAKE_CURRENT_LIST_DIR` for this use case.

Reply:
So I applied this change to the current PR:

```diff
diff --git a/cmake/FindPostgreSQL.cmake b/cmake/FindPostgreSQL.cmake
index 6b37a1a1f..feb9cc339 100644
--- a/cmake/FindPostgreSQL.cmake
+++ b/cmake/FindPostgreSQL.cmake
@@ -54,6 +54,9 @@
 get_filename_component(PRESENT_DIR "${CMAKE_CURRENT_LIST_FILE}" DIRECTORY)
 
 include(${PRESENT_DIR}/cmake/OpenSSL.cmake OPTIONAL RESULT_VARIABLE file_found)
+if(NOT ${PRESENT_DIR} STREQUAL ${CMAKE_CURRENT_LIST_DIR})
+    message(FATAL_ERROR "${PRESENT_DIR} but ${CMAKE_CURRENT_LIST_DIR}")
+endif()
 
 if(NOT DEFINED PG_CONFIG)
 
```

Then, I ran `make` for a global build as well as `ci/build-extensions-separately`, and none of them failed with the above error. 

To me, this suggests that `CMAKE_CURRENT_LIST_DIR` is good here. Am I missing anything you might have found?

Reply:
You are right. I think I got confused with the definition of `CMAKE_CURRENT_LIST_DIR`. Updated using it.